api/versions.go

```diff
@@ -365,3 +383,59 @@ func convertSpec(specV1 info.ContainerSpec) v2.ContainerSpec {
 	}
 	return specV2
 }
+
+func convertStats(cont *info.ContainerInfo) []v2.ContainerStats {
+	stats := []v2.ContainerStats{}
+	for _, val := range cont.Stats {
+		stat := v2.ContainerStats{
+			Timestamp:     val.Timestamp,
+			HasCpu:        cont.Spec.HasCpu,
+			HasMemory:     cont.Spec.HasMemory,
+			HasNetwork:    cont.Spec.HasNetwork,
+			HasFilesystem: cont.Spec.HasFilesystem,
+			HasDiskIo:     cont.Spec.HasDiskIo,
+		}
+		if stat.HasCpu {
+			stat.Cpu = val.Cpu
+		}
+		if stat.HasMemory {
+			stat.Memory = val.Memory
+		}
+		if stat.HasNetwork {
+			// TODO(rjnagal): Return stats about all network interfaces.
+			stat.Network = append(stat.Network, val.Network)
+		}
+		if stat.HasFilesystem {
+			stat.Filesystem = val.Filesystem
+		}
+		if stat.HasDiskIo {
+			stat.DiskIo = val.DiskIo
+		}
+		// TODO(rjnagal): Handle load stats.
+		stats = append(stats, stat)
+	}
+	return stats
+}
+
+func getStatsRequest(id string, r *http.Request) (v2.StatsRequest, error) {
```

Comment:
Used id here as it won't always be the name once we extend options to include docker.